cap/supervisor.go

```diff
@@ -10,19 +10,60 @@ import (
 	"github.com/capatazlib/go-capataz/internal/c"
 )
 
+// terminationManager offers an API to do thread-safe tracking of the
+// termination state of a Supervisor. This record is designed to hide the
+// required internal mutability on the Supervisor record.
+type terminationManager struct {
+	mux          *sync.Mutex
+	terminated   bool
+	terminateErr error
+}
+
+// newTerminationManager creates a new terminationManager
+func newTerminationManager() *terminationManager {
+	var mux sync.Mutex
+
+	return &terminationManager{
+		mux:          &mux,
+		terminated:   false,
+		terminateErr: nil,
+	}
+}
+
+// getTerminateErr is a (concurrent-safe) function that allows us to query if a
+// Supervisor was terminated.
+func (tm *terminationManager) getTerminateErr() (bool, error) {
+	tm.mux.Lock()
+	defer tm.mux.Unlock()
+	if tm.terminated {
+		// we already did the termination, just return the result
+		return tm.terminated, tm.terminateErr
```

Comment:
This should be the only line in this function, I think? Unless you want to check for the broken `!tm.termimated && tm.terminatedErr != nil`

Reply:
Yes, I think the `terminated` variable must be `true` for `terminateErr` to be filled... I think this is enough